docs/INSTALLATION.md

```diff
@@ -1,65 +1,90 @@
+## Install Stack
 
-```bash
-stack ghc
-```
+For doing Exercism exercises you only need Stack.
+
+To install Stack, simply run the GHCup installation command or script and follow the defaults but specify that you would like to have Stack installed as well.
+
+Did you miss the opportunity?
+No worries: just run `ghcup install stack` afterwards.
+
+GHCup will install
+
+- **A compiler** (GHC: the Glasgow Haskell Compiler)
+
+  Strictly speaking you do not need this for Exercism, but it comes in handy when you want to play with GHCi outside of exercise directories.
+
+- **Cabal**
+
+  A project/package manager.
+  You do not need this for Exercism.
+
+- **Stack**
 
-If this outputs something like `ghc: no input files`, GHC is installed
-correctly.
+  A project/package manager that you do need for Exercism.
+  It will make sure you have the right compiler and libraries for each individual project (exercise).
 
-Note that there are other ways to install GHC on a system (like the
-[Haskell platform](https://www.haskell.org/platform/)), but for
-running the Exercism tests, a regular stack installation is most
-likely to work correctly.
+To check that installation was successful, run the command `ghcup list` and/or `stack --version`.
 
-## Installing HLint (optional)
 
-HLint is a tool for suggesting possible improvements to Haskell code. These
-suggestions include ideas such as using alternative functions, simplifying
-code and spotting redundancies.
+## Get the right GHC version
 
-You can use Stack to install HLint:
+Stack will automatically download and install GHC when it is required.
+You can start the installation and test by running the following command in a terminal:
 
 ```bash
-stack install hlint
+# move into exercise directory
+cd /exercism/haskell/hello-world
+# start interactive environment
+stack ghc -- --version
 ```
 
-Check that it was installed correctly with:
+If this outputs something like
 
-```bash
-hlint --version
 ```
+The Glorious Glasgow Haskell Compilation System, version 8.8.4
+```
+
+then GHC is installed correctly.
+
+
+## Set up your editor to work with Haskell
+
+Haskell support is provided by the Haskell Language Server (HLS).
+Language support is not required, but strongly recommended.
+
+
+### Visual Studio Code
+
+Just get the [Haskell extension for Visual Studio Code][vscode-haskell].
+It will use GHCup to install HLS automatically, after which you'll be all set.
+
+**Note:** with freshly downloaded exercises, HLS may have trouble starting up.
+If this happens, just run `stack test` and restart the language server.
+
 
-If you get the version in your terminal, you're done, otherwise, you
-probably need to add a missing directory to your path:
+### Other editors
 
-- `%APPDATA%\local\bin`, if you are running *Windows*
-- `$HOME/.local/bin`, for other operating systems.
+See the [HLS documentation on editor configuration][hls-editor-config].
+You might need to install HSL manually.
```

Comment:
```suggestion
You might need to install HLS manually.
```

Of course, HSL would be hue-sat-lum, the notation for specifying colours. We need HLS (Haskell Language Server) here